### PR TITLE
fence_gce: Adds cloud-platform scope for bare metal API and optional proxy flags

### PR DESCRIPTION
**Pull request by kj1724**
This change adds the necessary scope for using the fence agent on bare metal. Along with that we have added proxy flags in case the customer is using a proxy for internet access instead of a NAT gateway

### CODE REVIEW

File: agents/gce/fence_gce.py [hunk 1]
```diff
@@ -9,13 +9,16 @@
 #
 
 import atexit
+import httplib2
 import logging
 import json
 import re
 import os
 import socket
+import socks
```
Comment (oalbrigt) [inline agents/gce/fence_gce.py:18]: You can move this down to the `try/except: pass` below, and you should be able to remove PySocks from `.travis.yml`.

**Review by oalbrigt — CHANGES_REQUESTED** [review]
Can you prefix the commits with `fence_gce: `?

File: agents/gce/fence_gce.py (hunk 1 above)
Comment (kj1724) [reply agents/gce/fence_gce.py]: I moved the import and updated the travis.yml file

**Comment by oalbrigt** [comment]
Will the bare metal scope work for every type of VM/bare metal, or should we make a parameter to be able to change it?

**Comment by oalbrigt** [comment]
ok to test

**Review by oalbrigt — APPROVED** [review]

**Comment by kj1724** [comment]
This bare metal scope work will allow the fence_gce agent to use the reset API for Google cloud VM/bare metal instances. We shouldn't need an extra parameter for now.

Thanks for reviewing!

**Comment by oalbrigt** [comment]
Thanks.

**Pull request merged**